api/response.go

```diff
@@ -36,15 +36,10 @@ func (r *Response) Decode(inStruct interface{}) (*Error, error) {
 	// TODO Remove as we'll use the common err format for this
 	if r.resp.StatusCode == 403 {
 		// Nothing to be done
-		return &Error{
-			Status:  http.StatusForbidden,
-			Message: "Forbidden",
-		}, nil
+		return ErrPermissionDenied, nil
 	}
 
-	apiErr := &Error{
-		Status: int32(r.resp.StatusCode),
-	}
+	apiErr := &Error{}
```

Comment:
Small nit: it seems the need for this variable falls away with these changes.  It took me some time to figure out exactly what we are doing with it here and perhaps a slightly more readable option:
1) on line 54 `instruct = &Error{}` // perhpas with a comment, but ok if not
2) on line 70 `return nil, &Error{response: r}`